src/test/scala/chiselTests/StrongEnum.scala

```diff
@@ -164,6 +164,38 @@ class StrongEnumFSM extends Module {
   }
 }
 
+object Opcode extends ChiselEnum {
+  val load = Value(0x03.U)
+  val imm = Value(0x13.U)
+  val auipc = Value(0x17.U)
+  val store = Value(0x23.U)
+  val litValues = List(0x03.U, 0x13.U, 0x17.U, 0x23.U)
+}
+
+class PrintableExecutionTest extends BasicTester {
+  val (count, done) = Counter(true.B, 5)
+  val w = WireDefault(Opcode.load)
+  when(count === 0.U) {
+    w := Opcode.load
+  }
+  when(count === 1.U) {
+    w := Opcode.imm
+  }
+  when(count === 2.U) {
+    w := Opconde.auipc
+  }
+  when(count === 3.U) {
+    w := Opcode.store
+  }
+  when(count === 4.U) { // invalid state
+    w := Opcode(Opcode.litValues.last + 0x1.U)
+  }
+  when(done) {
+    stop()
+  }
+  printf(cf"$w ")
```

Comment:
what happens if we print an illegal value of the `Opcode`?

Reply:
It currently doesn't print anything other than a warning in the log..

Reply:
should it print as "ILLEGAL VALUE" or something? Like:

cf"My Opcode is ${someOpcode}" should print "My Opcode is ILLEGAL VALUE" if it's 0.U in this example?

Reply:
Doesn't it print `?????`?

Reply:
If you had it print `ILLEGAL VALUE`, that would whitespace buffer the legal values of anything that takes fewer characters so I would vote against that.

Reply:
> Doesn't it print ??????

That would be cool I would want to see that in the testcase